packages/account/src/BiconomySmartAccount.ts

```diff
@@ -0,0 +1,173 @@
+    super({
+      bundlerUrl,
+      epAddress
+    })
+    this.provider = new JsonRpcProvider(rpcUrl);
+    this.entryPoint = EntryPoint_v100__factory.connect(epAddress, this.provider);
+    this.factory = SmartAccountFactory_v100__factory.connect(
+      factoryAddress,
+      this.provider
+    );
+    this.nodeClient = new NodeClient({ txServiceUrl: nodeClientUrl ?? NODE_CLIENT_URL })
+    try {
+      if (Signer.isSigner(signerOrProvider)) {
+        this.signer = signerOrProvider
+      } else if (Provider.isProvider(signerOrProvider)) {
+        this.signer = signerOrProvider.getSigner()
+      }
+      else {
+        console.error('signer or provider is not valid')
+      }
+    } catch (error) {
+      throw new Error("No signer provided")
+    }
+    console.log(paymasterUrl);
+    if (paymasterUrl)
+      this.paymaster = new BiconomyPaymasterAPI(paymasterUrl)
+  }
+  /**
+   * @description This function will initialise BiconomyAccount class state
+   * @returns Promise<BiconomyAccount>
+   */
+  async init(): Promise<BiconomySmartAccount> {
+    const instance = new BiconomySmartAccount({
+      signerOrProvider: this.biconomySmartAccountConfig.signerOrProvider,
+      rpcUrl: this.biconomySmartAccountConfig.rpcUrl,
+      epAddress: this.biconomySmartAccountConfig.epAddress,
+      factoryAddress: this.biconomySmartAccountConfig.factoryAddress,
+      bundlerUrl: this.biconomySmartAccountConfig.bundlerUrl,
+      paymasterUrl: this.biconomySmartAccountConfig.paymasterUrl
+    }
+    );
+    try {
+      instance.chainId = await this.provider.getNetwork().then((net) => net.chainId)
+      await instance.getInitCode()
+      await instance.entryPoint.callStatic.getSenderAddress(instance.initCode);
+
+      throw new Error("getSenderAddress: unexpected result");
+    } catch (error: any) {
+      const addr = await instance.factory.getAddressForCounterFactualAccount(await instance.signer.getAddress(), ethers.BigNumber.from(0))
+      instance.proxy = await SmartAccount_v100__factory.connect(addr, instance.provider);
+    }
+
+    return instance
+  }
+
+  async getInitCode(): Promise<string> {
+    this.initCode = ethers.utils.hexConcat([
+      this.factory.address,
+      this.factory.interface.encodeFunctionData("deployCounterFactualAccount", [
+        await this.signer.getAddress(),
+        ethers.BigNumber.from(0),
+      ])
+    ])
+    return this.initCode
+  }
+  /**
+   * @description an overrided function to showcase overriding example
+   * @returns 
+   */
+  nonce(): Promise<BigNumber> {
+    console.log('Overiding example');
+    return this.proxy.nonce()
+  }
+  /**
+   * 
+   * @param to { target } address of transaction 
+   * @param value  represents amount of native tokens
+   * @param data represent data associated with transaction
+   * @returns 
+   */
+  execute(to: string, value: BigNumberish, data: BytesLike) {
```

Comment:
From the implementation, this function name should be `getExecuteCallData`
Same for `executeBatch` method.

Reply:
resolved